### ISSUE TITLE
Could not find a suitable provider for docker_compose (when run from cron)

### ISSUE THREAD

**Issue by zahaim** [garethr/garethr-docker#466]
Hi,

Ubuntu 14.04, puppet 3.8.6, docker 1.9.1, docker-compose 1.5.2

My manifest:
```
    docker_compose { "$compose_dest_dir/$compose_file":
      ensure => present,
    }
```

I run puppet agent -t from cron and the result is:
```
puppet-agent[28092]: Could not find a suitable provider for docker_compose
```
When I run with debug:
```
puppet-agent[23914]: Puppet::Type::Docker_compose::ProviderRuby: file docker-compose does not exist
puppet-agent[23914]: Could not find a suitable provider for docker_compose
```

After adding to cron:
```
export PATH=$PATH:/usr/local/bin ; puppet agent -t
```
It works.

I resolved it by adding a link:
```
/bin# ln -s /usr/local/bin/
```
It would be nice to have it solved on module level I guess.

**Comment by jmverges** [comment]
Hi there, 

I'm having the same error output. any news regarding this?

**Comment by igrcic** [comment]
Hello there,

shouldn't the docker-compose also be installed together with this module? As it already provides docker_compose type. 

At first I didn't know one has to  install docker_compose seperately, so i got the same error.

**Comment by jmverges** [comment]
Same here @igrcic

**Comment by igrcic** [comment]
Ahh, sorry i haven't seen there is also a docker::compose class supplied. So one should just ensure it is installed  before starting the containers with docker-compose.

class { 'docker::compose': 
        ensure => present
} -> 
docker_compose { '/path/to/docker.yml':
     ensure  => present,
 }

**Comment by AblionGE** [comment]
Hi there,

It still does not work for me...
Here are the versions used:

* ```Docker version 1.11.2, build b9f10c9```
* ```docker-compose version 1.7.1, build 0a9ab35```

I'm installing everything in a Vagrant box (CentOS 7) and use ```puppet apply``` to apply my configuration (I do not use ```cron```).

I observed that the user ```root``` does not have ```/usr/local/bin/``` in his path and user ```vagrant``` has it. Thus, I added the path with ```panaman/pathmunge``` module and do the trick of the symbolic link. Nothing works... Has someone any idea on what can be the problem?

**Comment by igrcic** [comment]
Hello,

what does puppet agent -av say?

**Comment by AblionGE** [comment]
```
[root@localhost puppet]# puppet agent -av
Error: Could not parse application options: ambiguous option: -av
```

With ```--version``` I get ```3.8.7```.

As explained, I do not use the agent but ```puppet apply```. 
If I run ```docker-compose``` manually it works so it is not a problem in the installation of ```docker-compose```.

**Comment by patagona-afriemann** [comment]
@AblionGE 
I would assume this is due to /usr/local/bin not being in your $PATH.
Would it be possible to adapt the ruby resource to look for the specific path? Alternatively we could specify the install path to avoid this problem.

**Comment by AblionGE** [comment]
This is probably the issue. I will try to find a quick fix.

**Comment by BarnumD** [comment]
In my case, I had docker-compose installed, and /usr/local/bin already in the path on Ubuntu 18.04.  However, I had failed to install docker with `include 'docker'`.
